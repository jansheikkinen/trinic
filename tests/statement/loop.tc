// Loop

function main(void) -> void do
  loop print(1); end
  loop print(1); loop print(2); end end
end
