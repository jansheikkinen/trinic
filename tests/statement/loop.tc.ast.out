
### AST GENERATION ###
Declaration FUNCTION  main
  Stmt LOOP
    Stmt BUILTIN print
      Expr LITERAL
        INT 1
  Stmt LOOP
    Stmt BUILTIN print
      Expr LITERAL
        INT 1
    Stmt LOOP
      Stmt BUILTIN print
        Expr LITERAL
          INT 2
